### PR TITLE
Adds search sessions feature to Android

### LINKED ISSUE #305: Search option
It would be nice to search talks (title and description) and/or speakers (name and description).

### PR DESCRIPTION
**Pull request by marcellogalhardo**
A naive implementation of search sessions. The search component is simple but does look nice - and includes a transition animation. The same approach can be extended to speakers.

Fixes #305

[search-1.webm](https://user-images.githubusercontent.com/4348197/227663902-74ba721a-ab13-404a-a175-21d371330137.webm)

### CODE REVIEW

File: shared/src/mobileMain/kotlin/dev/johnoreilly/confetti/SessionsViewModel.kt [hunk 1]
```diff
@@ -147,6 +156,25 @@ open class SessionsViewModel : KMMViewModel(), KoinComponent {
...
+    fun onSearch(search: String) {
+        _search.value = search
+    }
+
+    private fun filterSessions(uiState: SessionsUiState, filter: String): SessionsUiState {
+        return if (filter.isNotBlank() && uiState is SessionsUiState.Success) {
+            val newSessions = uiState.sessionsByStartTimeList.map { outerMap ->
+                outerMap.mapValues { (_, value) ->
+                    value.filter { session ->
+                        session.title.contains(filter, ignoreCase = true)
```
Comment (martinbonnin) [inline shared/src/mobileMain/kotlin/dev/johnoreilly/confetti/SessionsViewModel.kt:168]: Might be worth searching the description as well?

**Review by martinbonnin — APPROVED** [review]
Nice!

File: shared/src/mobileMain/kotlin/dev/johnoreilly/confetti/SessionsViewModel.kt (hunk 1 above)
Comment (marcellogalhardo) [reply shared/src/mobileMain/kotlin/dev/johnoreilly/confetti/SessionsViewModel.kt]: Done. I have included the title, description, room, and speakers.

Comment (martinbonnin) [reply shared/src/mobileMain/kotlin/dev/johnoreilly/confetti/SessionsViewModel.kt]: Thanks!

**Pull request merged**